pandas/core/generic.py

```diff
@@ -5906,6 +5906,95 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    def as_nullable_dtypes(
+        self: FrameOrSeries, keep_integer: bool_t = False
+    ) -> FrameOrSeries:
+        """
+        Convert columns of DataFrame or a Series to types supporting ``pd.NA``.
+
+        If the dtype is "object", convert to "string", "boolean" or an appropriate
```

Comment:
If we are going to have this method, then we should deprecate infer_objects

Reply:
Not sure about that, because `infer_objects` converts to numpy types, which people may still want to use. I think we may want to deprecate `infer_objects` once all of the extension types are then the default ones.

Reply:
resolving this as the discussion is below